src/components/FilterWidget/FilterWidget.test.tsx

```diff
@@ -0,0 +1,225 @@
+
+  const origDispatch = store.dispatch;
+  store.dispatch = jest.fn(origDispatch);
+
+  const wrapper = rtlRender(
+    <Provider store={store}>
+      <AppContext.Provider
+        value={
+          {
+            modalContainer: '#modal-root',
+          } as any
+        }
+      >
+        <FilterWidget {...props} />
+      </AppContext.Provider>
+    </Provider>
+  );
+
+  return {
+    props,
+    store,
+    wrapper,
+  };
+};
+
+beforeEach(() => {
+  let modalRoot = document.getElementById('modal-root');
+  if (!modalRoot) {
+    modalRoot = document.createElement('div');
+    modalRoot.setAttribute('id', 'modal-root');
+    document.body.appendChild(modalRoot);
+  }
+});
+
+test('should render selected target property', () => {
+  const widgetId = '@widget/01';
+
+  const state = {
+    widgets: {
+      items: {
+        [widgetId]: {
+          isActive: false,
+          data: null,
+          widget: {
+            id: widgetId,
+            settings: {
+              targetProperty: 'geo_info.city',
+            },
+          },
+        },
+      },
+    },
+  };
+  const {
+    wrapper: { getByRole },
+  } = render({ ...state }, { id: widgetId });
+
+  const heading = getByRole('heading');
+  expect(heading.textContent).toEqual('city');
+});
+
+test('should render applied properties number', () => {
+  const {
+    wrapper: { getByTestId },
+  } = render();
+  const appliedPropertiesNumber = getByTestId('applied-properties-number');
+  expect(appliedPropertiesNumber.textContent).toEqual('1');
+});
+
+test('should fetch available property values if filter data is empty', () => {
+  const widgetId = '@widget/02';
+  const state = {
+    widgets: {
+      items: {
+        [widgetId]: {
+          isActive: false,
+          data: null,
+          widget: {
+            id: widgetId,
+            settings: {
+              targetProperty: 'geo_info.city',
+            },
+          },
+        },
+      },
+    },
+  };
+  const {
+    wrapper: { getByTestId },
+    store,
+  } = render({ ...state }, { id: widgetId });
+
+  const filterWidget = getByTestId('filter-widget');
+  fireEvent.click(filterWidget);
+  expect(store.dispatch).toHaveBeenCalledWith({
+    payload: { widgetId: widgetId },
+    type: SET_FILTER_WIDGET,
+  });
+});
+
+test('should not fetch property values when property list inside filter is not empty', () => {
+  const {
+    wrapper: { getByTestId },
+    store,
+  } = render();
+
+  const filterWidget = getByTestId('filter-widget');
+  fireEvent.click(filterWidget);
+  expect(store.dispatch).not.toHaveBeenCalled();
+});
+
+test('should render property values when property list inside filter is not empty', () => {
+  const {
+    wrapper: { getByTestId },
+  } = render();
+
+  const filterWidget = getByTestId('filter-widget');
+  fireEvent.click(filterWidget);
+  const scrollWrapper = getByTestId('scroll-wrapper');
+  const childElementCount =
+    scrollWrapper.children[0].children[0].childElementCount;
```

Comment:
Please, consider using WAI Aria roles to get the length of specific children (_getByRole_).